codegen-lite/src/main/java/software/amazon/awssdk/codegen/lite/regions/ServiceMetadataGenerator.java

```diff
@@ -214,16 +221,34 @@ private MethodSpec signingRegion() {
                          .build();
     }
 
-    private Map<String, Service> getServiceData(Partitions partitions) {
-        Map<String, Service> serviceData = new HashMap<>();
+    private Map<Partition, Service> getServiceData(Partitions partitions) {
+        Map<Partition, Service> serviceData = new TreeMap<>(Comparator.comparing(Partition::getPartition));
+
         partitions.getPartitions()
                   .stream()
                   .forEach(p -> p.getServices()
                                  .entrySet()
                                  .stream()
                                  .filter(s -> s.getKey().equalsIgnoreCase(service))
-                                 .forEach(r -> serviceData.put(p.getPartition(), r.getValue())));
+                                 .forEach(s -> serviceData.put(p, s.getValue())));
 
         return serviceData;
     }
+
+    /**
+     * Determines if a given region string is a "valid" AWS region.
+     *
+     * The region string must either match the partition regex, end with fips
+     * and match the partition regex with that included, or include the word "global".
+     *
+     * @param regex - Regex for regions in a given partition.
+     * @param region - Region string being checked.
+     * @return true if the region string should be included as a region.
+     */
+    private boolean validRegion(String regex, String region) {
+        return region.matches(regex) ||
+               region.endsWith("-fips") &&
+               region.matches(regex.replace("$", "(-fips)*$")) ||
```

Comment:
Have we verified how this affects existing regions?

Reply:
I've verified with the services that I know had issues that this works. 

It does however introduce some backwards incompatibilies.

Regions like sandbox and local are gone. Those are nice to have but we'd have to one off them.

I don't want to but we may want to have boths -fips as a suffix and fips- as a prefix. S3 has fips- as a suffix and we are removing those...